hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/operators/json/HazelcastJsonValueFunction.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.validate.operators.json;
+
+import com.hazelcast.jet.sql.impl.validate.HazelcastCallBinding;
+import com.hazelcast.jet.sql.impl.validate.operand.TypedOperandChecker;
+import com.hazelcast.jet.sql.impl.validate.operators.common.HazelcastFunction;
+import com.hazelcast.jet.sql.impl.validate.operators.typeinference.JsonFunctionOperandTypeInference;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.SqlCallBinding;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlJsonValueReturning;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlOperandCountRange;
+import org.apache.calcite.sql.SqlOperatorBinding;
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlReturnTypeInference;
+import org.apache.calcite.sql.type.SqlTypeName;
+
+@SuppressWarnings("checkstyle:MagicNumber")
+public class HazelcastJsonValueFunction extends HazelcastFunction {
+    public static final HazelcastJsonValueFunction INSTANCE = new HazelcastJsonValueFunction();
+
+    public HazelcastJsonValueFunction() {
+        super(
+                "JSON_VALUE",
+                SqlKind.OTHER_FUNCTION,
+                new JsonValueFunctionReturnTypeInference(),
+                new JsonFunctionOperandTypeInference(),
+                SqlFunctionCategory.SYSTEM
+        );
+    }
+
+    @Override
+    protected boolean checkOperandTypes(final HazelcastCallBinding callBinding, final boolean throwOnFailure) {
+        return JsonFunctionUtil.checkJsonOperandType(callBinding, throwOnFailure, 0)
+                && TypedOperandChecker.VARCHAR.check(callBinding, throwOnFailure, 1);
+    }
+
+    @Override
+    public SqlOperandCountRange getOperandCountRange() {
+        return SqlOperandCountRanges.between(2, 8);
```

Comment:
This call has 10 arguments:
```
JSON_VALUE(this, '$' returning tinyint DEFAULT 1 ON EMPTY default 1 on error)
```
we also need a test for this case. It's also weird, that we throw `Was expecting 2 arguments`.

I think we should use `SqlOperandCountRanges.from(2)` here. For example, this expression is parsed successfully:
```
JSON_VALUE(this, '$' default 1 on empty default 2 on empty)
```
The 2nd ON EMPTY clause overrides the 1st one. IMO we should throw an exception in this case. The user can provide any number of arguments and we should not say that there must be at most 10 arguments - the on-error and on-empty clauses don't look like operands, it's just how calcite handles it. BTW, calcite's `SqlJsonValueFunction` has range 2..10.

Reply:
We can probably solve this repeating clause issue with `JSON_VALUE` (since its implemented semi-manually), but not sure `JSON_QUERY` would be possible, its special handling is within Calcite's parts of analyzer, not yet sure how to override it actually.

Reply:
For `JSON_QUERY`, it's fine if we wait for calcite fix. It's not a serious issue.